doozer

```diff
@@ -1204,6 +1204,160 @@ def config_print(runtime, key, name_only):
+@click.option("--src-dest", "--sd", default="img-mirror.txt",
+              help="The SRC=DEST file to write to\ndefault=img-mirror.txt")
+@click.option("--image-stream", "--is", default="release-is.yaml",
+              help="The SRC=DEST file to write to\ndefault=release-is.yaml")
+@click.option("--is-base", "--isb", default=None,
+              help="The base image stream file up an empy '.spec.tags' array")
+@click.argument("pattern")
+@pass_runtime
+def release_mirror(runtime, src_dest, image_stream, is_base, pattern):
+    """Generates input for `oc` commands to mirror content and publish
+    image streams. One output is a SRC=DEST input for `oc image
+    mirror`, the other is an ImageStream YAML file used by `oc apply`.
+
+    Prints to STDOUT everything that is written to the output
+    files. This functions similar to how the `images:print` command
+    works. You MUST provide a format string for the SRC=DEST image
+    mirror PATTERN parameter. Multiple '=' characters are not allowed.
+
+    Allowed formatting keywords for PATTERN:
+
+    \b
+    {build} - Shorthand for {component}-{version}-{release} (e.g. container-engine-v3.6.173.0.25-1)
+    {component} - The component identified in the Dockerfile
+    {image_name_short} - The container image name without the registry (e.g. ose-ansible)
+    {image_name} - The container registry image name (e.g. openshift3/ose-ansible)
+    {image} - The image name in the Dockerfile
+    {name} - The name of the distgit repository (e.g. openshift-enterprise)
+    {namespace} - The image namespace ('containers' or 'rpms')
+    {release} - The release field in the Dockerfile
+    {repository} - Shorthand for {image}:{version}-{release}
+    {type} - The type of the distgit (e.g. rpms)
+    {version} - The version field in the Dockerfile
+
+    The output SRC=DEST file is simply a new line separated file where
+    the LHS is a source image and the RHS is the destination to mirror
+    the image to. For example:
+
+    \b
+        registry.reg-aws.openshift.com:443/{repository}=quay.io/openshift-release-dev/ocp-v4.0-art-dev:{version}-{release}-{image_name_short}
+
+    Eample base ImageStream YAML:
+
+    \b
+        kind: ImageStream
+        apiVersion: image.openshift.io/v1
+        metadata:
+          name: 4.0-art-latest
+          namespace: ocp
+        spec:
+          tags: []
+
+    """
+    runtime.initialize(clone_distgits=False)
+    images = [i for i in runtime.image_metas()]
+
+    # Load the ImageStream stub file
+    with open(is_base) as fp:
+        isb = yaml.safe_load(fp)
+
+    # All the items we will put in the image mirror input
+    src_dest_items = []
+
+    for image in images:
+        dfp = DockerfileParser(path=runtime.working_dir)
+        try:
+            dfp.content = image.fetch_cgit_file("Dockerfile")
+        except Exception:
+            click.echo("Error reading Dockerfile from distgit: {}".format(image.distgit_key))
```

Comment:
simply `raise DoozerFatalError(msg)` here instead. This will be caught at top level by doozer and instead of dropping a nasty stacktrace, will print the error in red and then `sys.exit(1)` cleanly.

Reply:
oh crap, lol

Reply:
I wonder if I meant to leave that there. Oh well. I'll follow your suggestion and see how it goes.